src/amber/cli/commands/pipelines.cr

```diff
@@ -0,0 +1,98 @@
+require "cli"
+require "shell-table"
+require "../helpers/sentry"
+
+module Amber::CLI
+  class MainCommand < ::Cli::Supercommand
+    class Pipelines < Command
+      getter result = Hash(String?, Array(String)).new
+      property current_pipe : String?
+
+      class Options
+        bool "--no-color", desc: "Disable colored output", default: false
+        bool "--no-plugs", desc: "Don't output the plugs", default: false
+        help
+      end
+
+      class BadRoutesException < Exception
+      end
+
+      ROUTES_PATH          = "config/routes.cr"
+      LABELS               = %w(Pipe Plug)
+      LABELS_WITHOUT_PLUGS = %w(Pipe)
+
+      PIPE_REGEX = /(pipeline)\s+\:(\w+)(?:,\s+\"([^\"]+)\")?/
+      PLUG_REGEX = /(plug)\s+([\w:]+)?/
```

Comment:
We've talked elsewhere about using extended regular expressions to add comments, clarity, and maintainability to our regular expressions. Can we do that here?

Reply:
@robacarp - I remember seeing your example in a previous PR, but I've never written an extended regex. Is this the idea?
```crystal
 /
   (plug)    # match and capture the word plug
   \s+       # followed by at least one whitespace character
   ([\w:]+)? # greedy optional match and capture one or more word characters or colons ':'
 /x
```

@NeverHappened - Does this description match what the regex is doing?

This was a fun little exercise, assuming the above is accurate, I would suggest a few changes:
```crystal
 /
   plug\s+  # match 'plug` and at least one whitespace character
   ([\w:]+) # match and capture one or more word characters or colons ':'
 /x
```

Reply:
I messed around a little more, WDYT of this:
```crystal
PLUG_REGEX = 
  /^         # start at beginning of line
    plug\s+  # match 'plug` and at least one whitespace character
    ([\w:]+) # match and capture one or more word characters or colons ':'
   $         # matches must come before end of line
  /x
```

See https://carc.in/#/r/3fhw

Reply:
@marksiemers that's the right idea! The only functional difference between regular regex and this is that whitespace is basically ignored outside of character classes, and that comments are allowed. I believe it adds a major maintainability component. :]

Reply:
Indeed, and it makes you think about what you are doing.

I used to teach at a bootcamp and we always told people - you'll never be able to read a regular expression. When you come back to it in 6 months, it will take 20 minutes to figure out - like starting from scratch. If only we had known about this!

Reply:
@marksiemers and @robacarp we can probably add a commit with these changes, this way we can have it merge sooner

Reply:
The pipeline one I'm giving up on for now, but you can see a WIP here: https://carc.in/#/r/3fiv

Reply:
If this works for the plug, I would say swap this in with this PR, as an example and a reminder:
```crystal
PLUG_REGEX = 
  /^         # start at beginning of line
    plug\s+  # match 'plug` and at least one whitespace character
    ([\w:]+) # match and capture one or more word characters or colons ':'
  $/x        # matches must come before end of line
```